From cb87fb548f57b2cb869b74f5b113afdd3eec45f0 Mon Sep 17 00:00:00 2001
From: Richard Gong <richard.gong@intel.com>
Date: Mon, 29 Mar 2021 16:03:00 -0500
Subject: [PATCH 24/41] HSD #22012814951-2: intel_fcs: replace __u32 with
 uint32_t

commit 9ef1287162ebd5b06ed1ec988be8a0b028e3001f from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.114-lts

__u32 is non-standard integer type so it is replaced by uint32_t.

Signed-off-by: Richard Gong <richard.gong@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 include/uapi/linux/intel_fcs-ioctl.h | 34 +++++++++-------------------
 1 file changed, 11 insertions(+), 23 deletions(-)

diff --git a/include/uapi/linux/intel_fcs-ioctl.h b/include/uapi/linux/intel_fcs-ioctl.h
index 66419850c1dc..a0483480d22a 100644
--- a/include/uapi/linux/intel_fcs-ioctl.h
+++ b/include/uapi/linux/intel_fcs-ioctl.h
@@ -37,8 +37,8 @@ enum fcs_certificate_test {
  * @rsvd: write as 0
  */
 struct intel_fcs_cert_test_word {
-	__u32	test_bit:1;
-	__u32	rsvd:31;
+	uint32_t test_bit:1;
+	uint32_t rsvd:31;
 };
 
 /**
@@ -53,7 +53,7 @@ struct intel_fcs_cert_test_word {
 struct fcs_validation_request {
 	enum fcs_vab_img_type so_type;
 	void *src;
-	__u32 size;
+	uint32_t size;
 };
 
 /**
@@ -63,7 +63,7 @@ struct fcs_validation_request {
  */
 struct fcs_key_manage_request {
 	void *addr;
-	__u32 size;
+	uint32_t size;
 };
 
 /**
@@ -76,8 +76,8 @@ struct fcs_key_manage_request {
 struct fcs_certificate_request {
 	struct intel_fcs_cert_test_word test;
 	void *addr;
-	__u32 size;
-	__u32 c_status;
+	uint32_t size;
+	uint32_t c_status;
 };
 
 /**
@@ -89,9 +89,9 @@ struct fcs_certificate_request {
  */
 struct fcs_data_encryption {
 	void *src;
-	__u32 src_size;
+	uint32_t src_size;
 	void *dst;
-	__u32 dst_size;
+	uint32_t dst_size;
 };
 
 /**
@@ -103,9 +103,9 @@ struct fcs_data_encryption {
  */
 struct fcs_data_decryption {
 	void *src;
-	__u32 src_size;
+	uint32_t src_size;
 	void *dst;
-	__u32 dst_size;
+	uint32_t dst_size;
 };
 
 /**
@@ -113,17 +113,7 @@ struct fcs_data_decryption {
  * @rndm: 8 words of random data.
  */
 struct fcs_random_number_gen {
-	__u32 rndm[8];
-};
-
-/**
- * struct fcs_version
- * @version: version data.
- * @flags: Reserved as 0
- */
-struct fcs_version {
-	__u32 version;
-	__u32 flags;
+	uint32_t rndm[8];
 };
 
 /**
@@ -154,7 +144,6 @@ struct intel_fcs_dev_ioctl {
 		struct fcs_data_encryption	d_encryption;
 		struct fcs_data_decryption	d_decryption;
 		struct fcs_random_number_gen	rn_gen;
-		struct fcs_version		version;
 	} com_paras;
 };
 
@@ -218,6 +207,5 @@ enum intel_fcs_command_code {
 #define INTEL_FCS_DEV_RANDOM_NUMBER_GEN \
 	_IOWR(INTEL_FCS_IOCTL, \
 	      INTEL_FCS_DEV_RANDOM_NUMBER_GEN_CMD, struct intel_fcs_dev_ioctl)
-
 #endif
 
-- 
2.18.1

